graph/src/blockchain/block_stream.rs

```diff
@@ -113,7 +113,7 @@ pub trait BlockStreamBuilder<C: Blockchain>: Send + Sync {
 
     async fn build_polling(
         &self,
-        chain: Arc<C>,
+        chain: &C,
```

Comment:
Nice, since this makes it clear that we are not holding on to the entire `Blockchain`